### PR TITLE
Feature/612

### PR DESCRIPTION
**Pull request by yuliiakucher**
Fixed admin confirmation form page. Added verification for the creating superadmin. When first superadmin wants to add second superadmin, he/she should firstly enter code that he/she will receive via email. 
Also superadmin can resend the email for the confirmation.

image
image
image

### Checklist
- [x] 🔽 My branch is up-to-date with "development" branch
- [x] ✅All tests passed locally
- [x] ✨My changes working with up-to-date front-end and admin part locally, like charm
- [x] 🔗 Link pull request to issue

### CODE REVIEW

File: src/pages/users/register-user/register-user.js [hunk 1]
```diff
@@ -29,28 +43,39 @@ const RegisterUser = ({ handleClose }) => {
...
   const formSchema = Yup.object().shape({
     email: Yup.string()
-      .required(loginErrorMessages.ENTER_EMAIL_MESSAGE)
-      .email(loginErrorMessages.INVALID_EMAIL_MESSAGE),
-    role: Yup.string().required(loginErrorMessages.SELECT_ROLE_MESSAGE)
+      .required(ENTER_EMAIL_MESSAGE)
+      .email(INVALID_EMAIL_MESSAGE),
+    role: Yup.string().required(SELECT_ROLE_MESSAGE),
+    code: Yup.string().when('role', {
+      is: 'superadmin',
```
Comment (AndriyLymych) [inline src/pages/users/register-user/register-user.js:56]: please, don't use magic strings

File: src/pages/users/register-user/register-user.js [hunk 2]
```diff
@@ -29,28 +43,39 @@ const RegisterUser = ({ handleClose }) => {
...
+      then: Yup.string().required(ENTER_CODE)
+    })
   });
 
   const { handleSubmit, handleChange, values, touched, errors } = useFormik({
     initialValues: {
-      role: '',
-      email: ''
+      role: 'admin',
```
Comment (AndriyLymych) [inline src/pages/users/register-user/register-user.js:63]: same

File: src/pages/users/register-user/register-user.js [hunk 3]
```diff
@@ -109,13 +138,44 @@ const RegisterUser = ({ handleClose }) => {
                 {touched.role && errors.role}
               </FormHelperText>
             </FormControl>
+            {values.role !== 'admin' && (
```
Comment (AndriyLymych) [inline src/pages/users/register-user/register-user.js:141]: same

File: src/pages/users/register-user/register-user.js [hunk 4]
```diff
@@ -109,13 +138,44 @@ const RegisterUser = ({ handleClose }) => {
...
+              <FormControl
+                className={styles.formControl}
+                error={touched.code && !!errors.code}
+              >
+                <Button
+                  id='send-code-button'
+                  data-cy='add-user-admin-button'
+                  onClick={handleSendCode}
+                  variant='contained'
```
Comment (AndriyLymych) [inline src/pages/users/register-user/register-user.js:150]: same

File: src/pages/users/register-user/register-user.js [hunk 5]
```diff
@@ -109,13 +138,44 @@ const RegisterUser = ({ handleClose }) => {
...
+                  className={styles.sendButton}
+                  color='primary'
```
Comment (AndriyLymych) [inline src/pages/users/register-user/register-user.js:152]: same

File: src/pages/users/register-user/register-user.js [hunk 6]
```diff
@@ -109,13 +138,44 @@ const RegisterUser = ({ handleClose }) => {
...
+                >
+                  {SEND_CODE}
+                </Button>
+                <TextField
+                  onChange={handleChange}
+                  value={values.code}
+                  id='code'
+                  variant='outlined'
+                  label='Код'
+                  name='code'
```
Comment (AndriyLymych) [inline src/pages/users/register-user/register-user.js:162]: same

File: src/pages/users/users-details/containers/user-details-card/user-details-card.js [hunk 7]
```diff
@@ -85,6 +87,16 @@ const UserDetailsCard = ({
             >
               {buttonStatus}
             </Button>
+            {buttonConfirmed && (
+              <Button
+                variant='contained'
+                color='primary'
```
Comment (AndriyLymych) [inline src/pages/users/users-details/containers/user-details-card/user-details-card.js:93]: same

File: src/redux/users/tests/users.saga.test.js [hunk 8]
```diff
@@ -187,7 +187,7 @@ describe('Users saga test', () => {
         [call(handleSuccessSnackbar, SUCCESS_CONFIRMATION_STATUS)]
       ])
       .put(setUsersLoading(false))
-      .put(push('/'))
+      .put(push('/login'))
```
Comment (AndriyLymych) [inline src/redux/users/tests/users.saga.test.js:190]: same

**Review by AndriyLymych — APPROVED** [review]

**Review by twihell — CHANGES_REQUESTED** [review]
When you create a new admin/superadmin and click "Створити", you can't see that your new user has been added to the general list of administrators. To see a newly added user, you have to reload the page.

1.
create-admin


2.
after clicking submit


3.
reload page

4.
after reload

**Review by AndriyLymych — APPROVED** [review]

**Review by twihell — APPROVED** [review]

**Pull request merged**